builder/openstack_id3/ssh.go

```diff
@@ -0,0 +1,72 @@
+package openstack_id3
+
+import (
+	"code.google.com/p/go.crypto/ssh"
+	"errors"
+	"fmt"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mitchellh/multistep"
+	"time"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/extensions/floatingip"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+)
+
+// SSHAddress returns a function that can be given to the SSH communicator
+// for determining the SSH address based on the server AccessIPv4 setting..
+func SSHAddress(compute_client *gophercloud.ServiceClient, sshinterface string, port int) func(multistep.StateBag) (string, error) {
+	return func(state multistep.StateBag) (string, error) {
+
+		s := state.Get("server").(*servers.Server)
+
+		if ip := state.Get("access_ip").(*floatingip.FloatingIP); ip.IP != "" {
+			return fmt.Sprintf("%s:%d", ip.IP, port), nil
+		}
+
+		// Does the pool actually exist?
+		if pool, ok := s.Addresses[sshinterface]; ok {
```

Comment:
This does not replicate the behavior of the old 'openstack' builder when 'ssh_interface' is unset (""). The old builder would loop through all addresses and choose the first available one.